### ISSUE TITLE
Feature Proposal: /watermark API endpoint for native SynthID text generation

### ISSUE THREAD

**Issue by kilopal** [guillaumemeyer/watermarks-remover#313]
Hey @guillaumemeyer, following up on our X thread! 

Proposing the design to implement a native `/watermark` endpoint. This gives us a built-in way to generate watermarked text for benchmarks without relying on the heavy `MarkLLM` harness. Per your note, I'm keeping this strictly scoped to the API for now.

### 1. API Endpoint Design
Following the routing pattern in `server.py`, we can expose `POST /watermark`. I noticed your `/clean` and `/inspect` endpoints use base64 encoding for the `"file"` payload. For this generation endpoint, we can accept raw strings for simplicity, or stick to base64 if you prefer strict consistency across the API. I've drafted it with raw strings below.

**Request:**
```json
{
  "text": "The raw text or prompt to be watermarked.",
  "keys": [118, 504, 421, 521], 
  "options": {
    "scheme": "synthid"
  }
}
```

**Response:**
```json
{
  "ok": true,
  "kind": "text",
  "watermarked_text": "...",
  "report": {
    "scheme_used": "synthid"
  }
}
```

### 2. Implementation Approach
* **Respecting the `stdlib-only` core:** Since `wr-core` is strictly Python 3.10+ stdlib-only, adding a logits processor directly would bloat the core requirements. Instead, we can design this as an optional backend/sidecar (similar to your `wr-synthid-score` setup) to keep the core service dependency-free while isolating the generation logic.
* **Batching Integration:** Wire this up to a `POST /watermark/batch` route to perfectly match your existing `/clean/batch` and `/inspect/batch` patterns.

Let me know if this payload structure and sidecar approach aligns with what you had in mind, and I can start putting together the PR!

**Comment by guillaumemeyer** [comment]
- Yes use a sidecar.
- Yes include batch
@kilopal lgtm, let's initiate a PR

**Comment by kilopal** [comment]
Awesome, thanks for the quick review! I'll get started on the sidecar architecture and the batch endpoints. I will open a Draft PR once the foundation is laid so we can make sure the implementation aligns.
